### ISSUE TITLE
Consider improving accessibility of the app

### ISSUE THREAD

**Issue by marcnause** [fossasia/pslab-android#3348]
**Actual Behaviour.**
Accessibility has not been a strong focus in the development of the PSLab app. E.g. I noticed that screen reader software stays silent when PSLab is used.

**Expected Behaviour**
The app meets the needs of visually impaired people with regard to accessibility.

We need top review the app, identify where it does not meet expectation yet and create sub-tickets for all issues which need to be fixed. If possible, we should also make sure that accessibility stays a focus in the future. Maybe it is possible to introduce automatic checks during CI.

**Additional context**
See https://docs.flutter.dev/ui/accessibility for a start.

**Would you like to work on the issue?**
Maybe.

**Comment by marcnause** [comment]
There are tools which can support accessibility checkin, e.g. https://pub.dev/packages/accessibility_tools

**Comment by Jashan32** [comment]
I went through the app using the ```accessibility_tools``` library and identified several accessibility issues:

- Some interactive elements (buttons/tap targets) have dimensions smaller than the recommended minimum touch target size of 48×48 dp.
- Multiple buttons and image widgets are missing semantic labels, making them difficult for screen readers such as ```TalkBack``` and ```VoiceOver``` to interpret.
- There are text scaling and responsive layout issues. When the system font size is increased, some text experiences overflow/clipping, resulting in content extending beyond its allocated bounds and reducing readability.

example:

<img height="550" alt="Image" src="https://github.com/user-attachments/assets/60df9354-355f-45e1-b899-b9bbf935d4de" />
